fix(docz-tools): Make Atlantis theme responsivise.

eddysims opened the pull request:
<!--
  Atlantis uses Conventional Commits to track versions.
  Pull request titles should follow the following format.

  For help creating your pull request, you can [use this tool](https://atlantis.getjobber.com/pull-request-name-generator)

  <TYPE>(<optional SCOPE>): <conditionally BREAKING CHANGE:> <description>

  eg.
    fix(SCOPE): stop graphite breaking when too much pressure applied — Patch Release
    feat(SCOPE): add 'graphiteWidth' option — (Minor) Feature Release
    feat(SCOPE): BREAKING CHANGE: remove graphiteWidth option — (Major) Breaking Release

  TYPE should consist of:
    - fix: a commit of the type fix patches a bug in your codebase
    - feat: a commit of the type feat introduces a new feature to the codebase
    - docs: documentation only changes
    - build: improvements to the build system
    - refactor: a change that neither fixes a bug nor introduces a feature
    - chore: other changes that don't modify src or test files

  SCOPE should be one of:
    - components
    - design
    - eslint
    - generators
    - hooks
    - stylelint


  If your pull request introduces a breaking change please append `BREAKING CHANGE:` following type / scope.

  Further Reading:
    - https://www.conventionalcommits.org
    - https://github.com/commitizen/conventional-commit-types/blob/master/index.json
-->

## Motivations

Atlantis is not very usable on a mobile device. This PR will make Atlantis work on a mobile device
 
## Changes

<!-- https://keepachangelog.com/en/1.0.0/ -->

### Added

- <!-- new features -->

### Changed

- Moved Layout component to use CSS modules
- Added responsive styles to Layout and Playground

### Deprecated

- <!-- soon-to-be removed features -->

### Removed

- <!-- now removed features -->

### Fixed

- <!-- for any bug fixes -->

### Security

- <!-- in case of vulnerabilities -->

## Testing

<!-- How to test your changes. -->

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

![Random photo of Atlantis](https://loremflickr.com/672/400/atlantis)

chris-at-jobber commented on packages/docz-tools/src/gatsby-theme-docz/components/Layout/Layout.tsx:
```diff
@@ -0,0 +1,61 @@
+    <>
+      <div className={styles.layout}>
+        <div className={styles.menu} style={menuStyle}>
+          <Button icon={open ? "remove" : "menu"} onClick={handleClick} />
```
can we make this a `tertiary` button?

chris-at-jobber commented:
@seanhealy can you take a look? Just the one comment from me

chris-at-jobber reviewed (changes requested):
just changing the toggle button type and I'm good

chris-at-jobber commented:
A question, non-blocking: is it possible to have the menu close when tapping the overlay?

chris-at-jobber commented on packages/docz-tools/src/gatsby-theme-docz/components/Layout/Layout.module.css:
```diff
@@ -0,0 +1,56 @@
+  z-index: 8888;
+  width: 100%;
+  height: 100vh;
+  background: var(--color-surface);
```
Another thought, also non-blocking... we should use a consistent color as with Menu and Modal (greyBlue at 80% opacity).

I was thinking as I did the color swaps in our components that I missed the opportunity to add a `color-overlay` variable, but looking at those components we actually implement in different ways. Menu uses an [rgba value](https://github.com/GetJobber/atlantis/blob/master/packages/components/src/Menu/Menu.css#L111) equivalent to greyBlue at 80%, while Modal uses [greyBlue and sets an opacity](https://github.com/GetJobber/atlantis/blob/master/packages/components/src/Modal/Modal.css#L29) on the element.

eddysims commented:
@chris-at-jobber I have addressed your comments.

Can you also take a look at `colors.css` and take a look at how I have approached the `--color-overlay`. I think that this is a good way to go with this, but would like your thoughts as well.

chris-at-jobber commented on packages/design/src/colors.css:
```diff
@@ -106,7 +108,8 @@
   --color-grey--dark: rgb(118, 118, 118);
 
-  --color-greyBlue: rgb(101, 120, 132);
+  --color--greyBlue--rgb: 101, 120, 132;
```
Ooooh this kind of blew my mind, I think this is good! I had to read it a few times before I realized _how_ it was working, but this is, I think, a good solution.

chris-at-jobber reviewed (approved):
Looks good to me! Might be good to have @seanhealy or @hosseintalebi take a look as well but this is a nice fix.
